arch/powerpc/kernel/eeh_driver.c

```diff
@@ -228,6 +228,7 @@ static void *eeh_report_error(void *data, void *userdata)
 
 	edev->in_error = true;
 	eeh_pcid_put(dev);
+	kobject_uevent(&dev->dev.kobj, KOBJ_OFFLINE);
```

Comment:
Try to use kobject_uevent_env instead. 

Like nobody uses KOBJ_OFFLINE/ONLINE, So use KOBJ_CHANGE and include a ptr to a buffer containing more info. 

Use kvm_uevent_notify_change as an example of this usage. 

KVM calls kvm_uevent_notify_change which in turn calls kobject_uevent_env. So all their uevents pass through their own uvent handler.

Reply:
Yes! will add it to next change.